File: InterviewHub/my-react-app/src/services/roleService.js
import axiosClient from './axiosClient';

export const roleService = {
  getAllRoles: () => {
    return axiosClient.get('/roles');
  },
  getRoleById: (id) => {
    return axiosClient.get(`/roles/${id}`);
  },
  createRole: (data) => {
    return axiosClient.post('/roles', data);
  },
  updateRole: (id, data) => {
    return axiosClient.put(`/roles/${id}`, data);
  },
  deleteRole: (id) => {
    return axiosClient.delete(`/roles/${id}`);
  }
};

File: Backend/src/main/java/com/InterviewHub/feature/role/RoleService.java
package com.InterviewHub.feature.role;

import com.InterviewHub.feature.role.dto.CreateRoleRequest;
import com.InterviewHub.feature.role.dto.RoleResponse;
import com.InterviewHub.feature.role.dto.UpdateRoleRequest;

import java.util.List;

public interface RoleService {
    RoleResponse createRole(CreateRoleRequest request);
    RoleResponse getRoleById(Long id);
    List<RoleResponse> getAllRoles();
    RoleResponse updateRole(Long id, UpdateRoleRequest request);
    void deleteRole(Long id);
}
